src/System/extract_system_rep.rs

```diff
@@ -46,7 +46,11 @@ pub fn create_executable_query<'a>(
 
                 let start_state: State = if let Some(state) = &**start {
                     validate_reachability_input(&machine, state)?;
-                    get_state(state, &machine, &transition_system)?
+                    let state = get_state(state, &machine, &transition_system)?;
+                    if state.get_location().id.is_partial_location() {
+                        return Err("Start state is a partial state, which it must not be".into())
```

Comment:
Is return required here?

Reply:
Yes, it is required, since we want to return a error if the start state is partial state. @mads256h

Reply:
I mean the keyword `return`

Reply:
Yes, it is required, since we want to return a error if the start state is partial state. @mads256h

Reply:
It is not the end of the function, so it will not return, it will just try setting `start_state` to Err which it cannot because it is a `State`.

Reply:
:+1: